bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttConnectionObserver.java

```diff
@@ -11,14 +11,18 @@
  * Implement this interface to get notified of connection state changes.
  * Register this observer at {@see MqttBrokerConnection}.
  *
+ * @author David Graeff - Rewritten
  * @author Markus Rathgeb - Initial contribution and API
  */
 public interface MqttConnectionObserver {
     /**
-     * Inform the observer if the connection is active or disconnected.
+     * Inform the observer if a connection could be established or if a connection
+     * is lost. This will be issued in the context of the Mqtt client thread and
+     * requires that the control is returned quickly to not stall the Mqtt thread.
      *
-     * @param connected true if the connection is established, false if disconnected.
+     * @param connected True if the connection is established, false if disconnected.
+     * @param error An exception object (might be a MqttException) with the reason why
+     *            a connection failed.
      */
-    public void setConnected(boolean connected);
-
+    public void connectionStateChanged(boolean connected, Throwable error);
```

Comment:
This is an API breaking change. If we get aways with this, how about "extending" it a little further by passing an enum `ConnectionState` with `CONNECTED, DISCONNECTED` instead of the boolean. This will allow us to extend the API w/o breaking it. wdyt?